pkg/specs/pvcs.go

```diff
@@ -131,54 +145,92 @@ func DetectPVCs(
 			continue
 		}
 
+		// TODO optimize
+		// we save all the instance we found so far
+		instanceName := pvc.Labels[utils.InstanceLabelName]
+		if !slices.Contains(foundInstances, instanceName) {
+			foundInstances = append(foundInstances, instanceName)
+		}
+		// END TODO
+
+		instancePVC := apiv1.InstancePVC{
+			InstanceName: instanceName,
+			PvcName:      pvc.Name,
+		}
+
 		if isResizing(pvc) {
-			result.Resizing = append(result.Resizing, pvc.Name)
+			result.Resizing = append(result.Resizing, instancePVC)
 		}
 
 		// Find a Pod corresponding to this PVC
-		podFound := false
 		for idx := range podList {
-			if podList[idx].Name == pvc.Name {
-				podFound = true
-				break
+			for _, volume := range podList[idx].Spec.Volumes {
+				if volume.PersistentVolumeClaim != nil && volume.PersistentVolumeClaim.ClaimName == instancePVC.PvcName {
+					// We found a Pod using this PVC so this
+					// PVC is not dangling
+					result.Healthy = append(result.Healthy, instancePVC)
+					continue pvcLoop
+				}
 			}
 		}
 
-		if podFound {
-			// We found a Pod using this PVC so this
-			// PVC is not dangling
-			result.Healthy = append(result.Healthy, pvc.Name)
-			continue
-		}
-
-		jobFound := false
 		for idx := range jobList {
 			if IsJobOperatingOnPVC(jobList[idx], pvc) {
```

Comment:
I think we need to improve the definition of `IsJobOperatingOnPVC` to look at the Job definition, like what we do in the previous loop. Maybe we could have a function doing exactly.